src/Core/Primitives.idr

```diff
@@ -65,33 +65,25 @@ castInt _ = Nothing
 castBits8 : Vect 1 (NF vars) -> Maybe (NF vars)
 castBits8 [NPrimVal fc (BI i)]
     = let max = prim__shl_Int 1 8 in
-          if i > 0 -- oops, we don't have `rem` yet!
-             then Just (NPrimVal fc (B8 (fromInteger i `mod` max)))
-             else Just (NPrimVal fc (B8 (max + fromInteger i `mod` max)))
+          Just (NPrimVal fc (B8 (fromInteger i `rem` max)))
```

Comment:
I think the comment was meant to indicate that `mod` was implemented as `rem`, thus the workaround. Can you change the code from `rem` to `mod`, now that you have fixed the implementation of `mod`? (Also for `castBits16`, `castBits32`, `castBits64`.)